tests/rbd/readonly_namespace_image_migration.py

```diff
@@ -0,0 +1,157 @@
+
+Test Case Flow :
+1. Create pool and an image
+2. Create a namespace within the pool
+3. Give readonly access to the namespace
+4. Create images within the namespace
+5. Migrate the created images to other pool
+6. verify for image migration
+"""
+
+from tests.rbd.exceptions import RbdBaseException
+from tests.rbd.rbd_utils import (
+    Rbd,
+    initial_rbd_config,
+    verify_migration_commit,
+    verify_migration_state,
+    verify_namespace_exist,
+)
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def image_migrate_and_verify(
+    rbd,
+    src_pool,
+    dest_pool,
+    namespace,
+    image,
+):
+    """Migrate the image from sorce to destination and verify migration
+
+    Args:
+        src_pool: source pool name
+        dest_pool: destination pool name
+        namespace: name of namespace
+    """
+    src_spec1 = src_pool + "/" + namespace + "/" + image
+    dest_spec1 = dest_pool + "/" + image
+    cmd1 = "ceph auth caps client.1"
+    cmd1 += " mon 'allow *' mgr 'allow profile rbd-read-only'"
+    cmd1 += f" osd 'profile rbd-read-only pool={src_pool}"
+    cmd1 += " namespace=testnamespace' -o /etc/ceph/ceph.client.1.keyring"
+    rbd.exec_cmd(cmd=cmd1)
+    rbd.create_image(
+        pool_name=src_pool,
+        image_name=image,
+        size="5G",
+        namespace=namespace,
+    )
+    rbd.image_exists(
+        pool_name=src_pool,
+        image_name=image,
+        namespace=namespace,
+    )
+    rbd.migration_prepare(src_spec1, dest_spec1)
+    verify_migration_state(rbd, dest_spec1)
+    rbd.migration_action(action="execute", dest_spec=dest_spec1)
+    verify_migration_state(rbd, dest_spec1)
+    rbd.migration_action(action="commit", dest_spec=dest_spec1)
+    verify_migration_commit(rbd, dest_pool, image=image)
+
+
+def run(**kw):
+    """verify for image migration when Set read only on namespace and migrate the images.
+
+    Args:
+        kw: test data
+
+    Returns:
+        int: The return value. 0 for success, 1 otherwise
+    """
+
+    for key, value in kw["config"].items():
+        if key == "source":
+            kw["config"].update(value)
+            initial_rbd_config(**kw)
+            for key in kw["config"]["source"]:
+                kw["config"].pop(key)
+            kw["config"].pop("ec-pool-k-m")
+        elif key == "destination":
+            kw["config"].update(value)
+            initial_rbd_config(**kw)
+            for key in kw["config"]["destination"]:
+                kw["config"].pop(key)
+            kw["config"].pop("ec-pool-k-m")
+
+    rbd = Rbd(**kw)
+    src_pools = []
+    src_pool1 = kw["config"]["source"]["rep_pool_config"]["pool"]
+    src_pools.append(src_pool1)
+    src_pool2 = kw["config"]["source"]["ec_pool_config"]["pool"]
+    src_pools.append(src_pool2)
+    dest_pool1 = kw["config"]["destination"]["rep_pool_config"]["pool"]
+    dest_pool2 = kw["config"]["destination"]["ec_pool_config"]["pool"]
+    namespace = "testnamespace"
```

Comment:
Input this value as well from kw["config"] and if not given set default as testnamespace

Reply:
Done.